addons/hr/models/resource.py

```diff
@@ -35,3 +36,16 @@ def _compute_avatar_128(self):
                 resource.avatar_128 = employee[0].avatar_128
             else:
                 resource.avatar_128 = avatar_per_employee_id[employee[0].id]
+
+class ResourceCalendar(models.Model):
+    _inherit = 'resource.calendar'
+
+    def _calculate_hours_per_week(self):
+        self.ensure_one()
+        sum_hours = sum(
+            (a.hour_to - a.hour_from) for a in self.with_context(from_hours_per_week=True)._get_global_attendances())
+        return sum_hours / 2 if self.two_weeks_calendar else sum_hours
+
+    def _calculate_is_fulltime(self):
+        self.ensure_one()
+        return not float_compare(self.full_time_required_hours, self._calculate_hours_per_week(), 3)
```

Comment:
First, I don't see why this should be in HR rather than resource.
Secondly, though it will be solved by moving the code, please define models inside files with a related name (`resource.calendar` → `resource_calendar.py`)